### PR TITLE
Warn on singular and plural configs

### PR DESCRIPTION
**Pull request by eatonphil**

### CODE REVIEW

File: src/api/apiResultActionReducerGenerator.js [hunk 1]
```diff
@@ -11,6 +11,12 @@ export const DELETE = 'DELETE';
  */
 export function genConfig(config, parent = undefined) {
   const result = { primaryKey: 'id', ...config, parent };
+  if (result.singular && result.plural) {
+    throw new Error(
+      'Config must not set "singular" _and_ "plural" keys. Plural is used for ' +
```
Comment (mparke) [inline src/api/apiResultActionReducerGenerator.js]: can we make this a comment in the function documentation, as opposed to throwing an error for incorrect implementation?

Comment (mparke) [inline src/api/apiResultActionReducerGenerator.js]: if this function is that difficult to use correctly, we should change the function

Comment (eatonphil) [inline src/api/apiResultActionReducerGenerator.js]: prs welcome :/

**Comment by eatonphil** [comment]
Will reopen after trying a macro system.

**Pull request closed without merging**